### PR TITLE
Iss 153

### LINKED ISSUE #153: Replace hardcoded velocity in high level control with real velocity from localization

### PR DESCRIPTION
**Pull request by menson168**
closes #153

### CODE REVIEW

**Comment by menson168** [comment]
Please check if I have referred to the correct component in the chassis_state message. I assumed that the linear velocity of the vehicle would be the velocity in the linear domain of X axis. @joeyzhu00

File: src/modules/control/src/control_node.cpp [hunk 1]
```diff
@@ -143,7 +143,8 @@ namespace node
       prev_pos[0] = position(0);
       prev_pos[1] = position(1);
     }*/
-
+    
+    linear_velocity = veh_state_msg.Speed.twist.linear.x;
```
Comment (joeyzhu00) [inline src/modules/control/src/control_node.cpp]: I forgot to mention that the Chassis_state message is in the odom/gps frame. This should be changed to:
`linear_velocity =  sqrt(veh_state_msg.Speed.twist.linear.x^2 + veh_state_msg.Speed.twist.linear.y^2);`

Comment (menson168) [reply src/modules/control/src/control_node.cpp]: Fixed

**Review by joeyzhu00 — APPROVED** [review]
Looks good!

**Comment by menson168** [comment]
@joeyzhu00  Now it is fixed.

**Review by joeyzhu00 — APPROVED** [review]

**Pull request merged**